tests/console/python3_pynacl.pm

```diff
@@ -0,0 +1,91 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Simple pynacl libsodium smoke test with transactional and non-transactional support
+#
+# Maintainer: QE Core <qe-core@suse.de>
+#
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use serial_terminal 'select_serial_terminal';
+use registration qw(runtime_registration);
+use version_utils;
+use python_version_utils;
+
+my $requires_scc_registration = is_sle_micro || is_sle;
+my $script_download_path = "~/test_pynacl.py";
+
+sub test_setup {
+    select_serial_terminal;
+    # Download the test script
+    assert_script_run("curl -v -o $script_download_path " . data_url("python/pynacl/test_pynacl.py"));
+
+    runtime_registration() if $requires_scc_registration;
+}
+
+sub run_test {
+    my ($python_package) = @_;    # Proper declaration of $python_package
+    my $pkg = "$python_package-PyNaCl";
+
+    # we don't run the test if PyNaCl is not packaged for this python version
+    return unless script_run("zypper se $pkg") == 0;
+    record_info("Testing for $python_package");
+
+    if (is_transactional) {
+        trup_call("pkg in $python_package $pkg");
+        process_reboot(expected_grub => 1, trigger => 1);
+        select_serial_terminal;
+    } else {
+        zypper_call("in $python_package $pkg");
+    }
+
+    my $python_interpreter = get_python3_binary($python_package);
+    record_info("running python version", script_output("$python_interpreter --version"));
+
+    # Run the Python test script with the appropriate Python version
+    my $output = script_output("$python_interpreter " . $script_download_path, proceed_on_failure => 0, type_command => 1);
```

Comment:
Why are you using `proceed_on_failure => 0, type_command => 1`
If it would fail then it would fail, then you don;ty need to do redundant `if ($output =~ /Message verified successfully/) {`